tasks.py

```diff
@@ -38,6 +39,95 @@
 yq_version = "v4.45.1"
 
 
+def load_test_filters():
+    """Load test filter configuration from YAML file."""
+    # Find the config file relative to this module
+    script_dir = Path(__file__).parent
+    config_path = script_dir / "e2etest" / "test-filters.yaml"
+
+    if not config_path.exists():
+        raise FileNotFoundError(f"Configuration file not found: {config_path}")
+
+    with open(config_path, "r") as f:
+        return yaml.safe_load(f)
+
+
+def generate_skip_patterns(bgp_type, ip_family, protocol=None, with_prometheus=None):
+    """
+    Generate skip patterns based on environment configuration.
+
+    Args:
+        bgp_type: BGP type (native, frr, frr-k8s, frr-k8s-external)
+        ip_family: IP family (ipv4, ipv6, dual)
+        protocol: Protocol type (bgp, layer2) - optional
+        with_prometheus: Whether Prometheus is present (bool) - optional
+
+    Returns:
+        List of skip pattern strings
+    """
+    config = load_test_filters()
+    filters = config["test_filters"]
+    skip_patterns = []
+
+    # Add BGP type skip patterns
+    if bgp_type in filters["bgp_type"]:
+        skip_patterns.extend(filters["bgp_type"][bgp_type]["skip"])
+
+    # Add IP family skip patterns
+    if ip_family in filters["ip_family"]:
+        skip_patterns.extend(filters["ip_family"][ip_family]["skip"])
+
+    # Add protocol skip patterns (if provided)
+    if protocol and protocol in filters["protocol"]:
+        skip_patterns.extend(filters["protocol"][protocol]["skip"])
+
+    # Add Prometheus skip patterns (if Prometheus is not available)
+    if with_prometheus is False:
+        if "prometheus" in filters and "skip" in filters["prometheus"]:
+            skip_patterns.extend(filters["prometheus"]["skip"])
+
+    # Check special cases and apply additional skip patterns
+    for special_case in filters.get("special_cases", []):
+        condition = special_case["condition"]
+
+        # Check if this special case applies by matching all condition fields
+        matches = True
+        for key, value in condition.items():
+            if key == "bgp_type" and bgp_type != value:
+                matches = False
+                break
+            elif key == "ip_family" and ip_family != value:
+                matches = False
+                break
+
+        # If condition matches, add the skip patterns
+        if matches:
+            if "skip" in special_case:
+                skip_patterns.extend(special_case["skip"])
```

Comment:
let's start with hardcoding the single "special case" that we have (which is native + ipv6) instead of this logic (and the part in the file). if/when it becomes necessary we can enhance